server/repository/articleRepository.js

```diff
@@ -110,6 +112,64 @@ class ArticleRepository {
     }
     return article;
   }
+
+  /**
+   * Adds an article to  a user's reading stats
+   * @param {object} user Reader
+   * @param {object} article Read article
+   * @returns {object} New or updated reading stat
+   */
+  static async addReadingStat(user, article) {
+    const hasRead = await user.hasReadArticle(article);
+
+    if (!hasRead) {
+      const newRead = await user.addReadArticle(article);
+      return newRead[0][0].dataValues;
+    }
+
+    const justUpdated = await ReadingStat.update(
+      { timeRead: new Date() },
```

Comment:
no need for this. the updatedAt column serves the same purpose

Reply:
The reason we did this is to have a field to update when performing the operation. The alternative would be to update the userId and articleId fields each time.